kernel/syscall.c

```diff
@@ -2671,8 +2671,26 @@ long myst_syscall_getsockname(
     myst_fdtable_t* fdtable = myst_fdtable_current();
     myst_sockdev_t* sd;
     myst_sock_t* sock;
+    myst_fdtable_type_t type;
+    void* device = NULL;
+    void* object = NULL;
+
+    if (!myst_is_good_addr(addrlen) || !myst_is_good_addr(addr))
+    {
+        ret = -EFAULT;
+        goto done;
+    }
+
+    ECHECK(myst_fdtable_get_any(fdtable, sockfd, &type, &device, &object));
```

Comment:
I think it's fine to just pass `sd` and `sock` as-is. Do you see any problem with it?

Reply:
yeah - it expects void**, and this is how other syscalls in this file were structured

Reply:
Given that you will casting them back anyway, you can cast them with (void**) here.
See https://github.com/deislabs/mystikos/blob/beba4ea27622a64d1698dfd1c9d997f391c59dd5/kernel/syscall.c#L2530

Reply:
I agree that you can just cast them an not have the temporaries.

Reply:
Ah, I was trying &((void*)sd) which failed and I see why, this makes more sense. Thank you!